terraform_provisioning/load_balancing_application/load_balancing_application.tf

```diff
@@ -0,0 +1,166 @@
+  protocol        = "TCP"
+  protocol_port   = 22
+  loadbalancer_id = openstack_lb_loadbalancer_v2.loadbalancer.id
+  timeout_client_data = 600000
+  timeout_member_connect = 600000
+  timeout_member_data = 600000
+}
+
+# Create an ssh listener for the web servers
+resource "openstack_lb_listener_v2" "web_server_listener" {
+  name = "web_servers"
+  protocol        = "TCP"
+  protocol_port   = 80
+  loadbalancer_id = openstack_lb_loadbalancer_v2.loadbalancer.id
+  timeout_client_data = 600000
+  timeout_member_connect = 600000
+  timeout_member_data = 600000
+}
+
+# Creating the ssh pool
+resource "openstack_lb_pool_v2" "ssh_pool" {
+  name = "ssh-pool"
+  protocol    = "TCP"
+  lb_method   = "SOURCE_IP"
+  listener_id = openstack_lb_listener_v2.ssh_listener.id
+}
+
+# Creating a web_servers pool
+resource "openstack_lb_pool_v2" "web_servers_pool" {
+  name = "web_servers_pool"
+  protocol    = "TCP"
+  lb_method   = "SOURCE_IP"
+  listener_id = openstack_lb_listener_v2.web_server_listener.id
+}
+
+# Adds the bastion to the ssh pool with the ssh listener
+resource "openstack_lb_member_v2" "ssh_member" {
+  pool_id       = openstack_lb_pool_v2.ssh_pool.id
+  address       = openstack_compute_instance_v2.bastion.access_ip_v4
+  protocol_port = 22
+}
+
+# Creating the web_server member
+resource "openstack_lb_member_v2" "web_server_member" {
+  pool_id       = openstack_lb_pool_v2.web_servers_pool.id
+  count = length(openstack_compute_instance_v2.vm)
+  address = openstack_compute_instance_v2.vm[count.index].access_ip_v4
+  protocol_port = 80
+}
+
+# Creating the keypairs for the vms
+resource "openstack_compute_keypair_v2" "vm-keypair" {
+  name = "vm-keypair"
+}
+
+# Creating the private key from the created keypair
+resource "local_file" "vm-keypair-private" {
+  connection {
+    type = "ssh"
+    user = "ubuntu"
+    host = openstack_compute_instance_v2.bastion.access_ip_v4
+    port = 22
+  }
+
+  filename = "/home/ubuntu/.ssh/id_rsa"
+  content = openstack_compute_keypair_v2.vm-keypair.private_key
+}
+
+# Copying the private key from the locally created
+/*
+resource "null_resource" "copying_private_key" {
+    connection {
+        type = "ssh"
+        user = "ubuntu"
+        host = openstack_compute_instance_v2.bastion.access_ip_v4
+        port = 22
+    }
+
+    provisioner "file" {
+        source = "id_rsa"
+        destination = "/home/ubuntu/.ssh/id_rsa"
+    }
+
+    depends_on = [ openstack_lb_member_v2.ssh_member ]
+}
+*/
+
+# Create multiple vm's for web serving
+resource "openstack_compute_instance_v2" "vm" {
+  count = var.instances
+  image_name = var.image_name
+  flavor_name = var.flavor_name
+  key_pair = openstack_compute_keypair_v2.vm-keypair.name
+  name = format("vm-%d", count.index)
+
+  network {
+    uuid = openstack_networking_network_v2.private_network.id
+  }
+
+  # Due to weird issue faced with terraform creating vm's before subnets are made
+  depends_on = [ openstack_networking_subnet_v2.subnet ]
```

Comment:
We should put a depends on in the LB member for these VMs too, else it will create LB listeners before the VMs come up